controllers/configmap_controller.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Perses Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	observabilityv1alpha1 "github.com/perses/perses-operator/api/v1alpha1"
+	common "github.com/perses/perses-operator/internal/perses/common"
+	subreconciler "github.com/perses/perses-operator/internal/subreconciler"
+	logger "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var cmlog = logger.WithField("module", "configmap_controller")
+
+func (r *PersesReconciler) reconcileConfigMap(ctx context.Context, req ctrl.Request) (*ctrl.Result, error) {
+	perses := &observabilityv1alpha1.Perses{}
+
+	if r, err := r.getLatestPerses(ctx, req, perses); subreconciler.ShouldHaltOrRequeue(r, err) {
+		return r, err
+	}
+
+	configName := common.GetConfigName(perses.Name)
```

Comment:
String manipulation are general expensive, since this is going to run every reconcile loop, maybe worth reuse the config name through the entire reconcile loop, send it as a parameter to `configMapForPerses` wdty?

Reply:
This will be called only when the config does not exists, if is there this function will be skipped.